int32[] botIds
bool[] status